src/main.c

```diff
@@ -397,146 +397,111 @@ static void apply_operations(numberstack* numbers, operation** current_op) {
-
-                        in[++pos] = '\0';
-                        len = pos;
+                if (browsing) {    // The right arrow should only work while in the middle of the input
+                    pos++;
+    
+                    // Exit browsing mode if the cursor is at the end of the input
+                    if (pos == len) {
                         browsing = 0;
                     }
                 }
                 searched = 1;
             }
         }
 
-        if (inp == 127)
-        {
+        if (inp == 127) {
             // Backspace
 
-            if (pos != 0){
+            if (pos != 0) {
                 pos--;
                 len--;
+                inp = '\0';
+            }
+            else {
+                // Skip printing if backspace was pressed but nothing was done, otherwise a strange undefined character is printed
+                searched = 1;
             }
-            inp = '\0';
         }
 
-        if(!searched) {
-            // Prevent user to input more than MAX_IN
-            if (len <= MAX_IN && !browsing) {
+        // Prevent user to input more than MAX_IN
+        if(!searched && len <= MAX_IN) {
+            if (!browsing) {
                 // If the cursor is at the end of the text
                 
                 // Append char to in array
                 in[pos] = inp;
                 in[++pos] = '\0';
                 len++; // Make sure that len is still equal to pos
 
-                if (inp == '\0')
-                {
+                if (inp == '\0') {
                     // Clear screen from previous input
                     mvwprintw(inputwin, 1, 22 + --len ," ");
                 }
             }
-            else if (len < MAX_IN && browsing) {
+            else {
                 // If the cursor is in the text, not at the end
-                // len < MAX_IN is allowed because the underscore adds one to length and the underscore isn't really a part of in
 
-                if (inp == '\0'){
+                if (inp == '\0') {
                     // Backspace
                     
-                    // Check if pos is already at the start of the input, if so don't do anything
-                    if (in[0] != '_'){
-                        in[pos] = '_';
-
-                        // Move all of in after pos one space back
-                        for (int i = pos; i <= len; i++){
-                            in[i] = in[i + 1];
-                        }
-                        // Clear screen from previous input
-                        mvwprintw(inputwin, 1, 22 + len ," ");
+                    // Move all of in after pos one space back
+                    for (int i = pos; i <= len; i++) {
+                        in[i] = in[i + 1];
                     }
+                    // Clear screen from previous input
+                    mvwprintw(inputwin, 1, 22 + len ," ");
```

Comment:
Can you put the space after the comma like this: `mvwprintw(inputwin, 1, 22 + len, " ");`